library/src/main/java/com/wultra/android/mtokensdk/api/operation/AttributeTypeAdapter.kt

```diff
@@ -94,18 +94,22 @@ internal class AttributeTypeAdapter : TypeAdapter<Attribute>() {
         fun build(): Attribute? {
             return when (type) {
                 Attribute.Type.AMOUNT -> {
-                    val amount: BigDecimal = attr("amount") ?: return null
-                    val currency: String = attr("currency") ?: return null
-                    AmountAttribute(amount, currency, attr("amountFormatted"), attr("currencyFormatted"), attr("valueFormatted"), label)
+                    // For backward compatibility with legacy implementation, where the `amountFormatted` and `currencyFormatted` values might not be present,
+                    // we decode from `amount` and `currency` which were not nullable
+                    val amountFormatted: String = attr("amountFormatted") ?: attr<String>("amount").toString()
+                    val currencyFormatted: String = attr("currencyFormatted") ?: attr("currency")!!
```

Comment:
This should be the same (or simular logic since currency and currencyFormated are both strings) as `amount`.